### PR TITLE
fix(controller): support parsing github enterprise urls

### LINKED ISSUE #2080: bug?: github prs may not be working for github enterprise
I know @jessesuen had intended for PRs to GitHub Enterprise to work, as evidenced by the predicate used when registering the GitHub implementation of the `GitProviderService` interface:

https://github.com/akuity/kargo/blob/2d4aaf579625fd723ea4b3941c226ec9bea23d5e/internal/gitprovider/github/github.go#L22-L32

However, this function is used several places throughout that implementation and appears that an error will result for any URL not beginning with `https://github.com`:

https://github.com/akuity/kargo/blob/2d4aaf579625fd723ea4b3941c226ec9bea23d5e/internal/gitprovider/github/github.go#L149-L156

### PR DESCRIPTION
**Pull request by krancour**
Fixes #2080

### CODE REVIEW

File: internal/gitprovider/github/github.go [hunk 1]
```diff
@@ -146,13 +145,18 @@ func convertGithubPR(ghPR *github.PullRequest) *gitprovider.PullRequest {
...
+func parseGitHubURL(repoURL string) (string, string, error) {
+	u, err := url.Parse(repoURL)
+	if err != nil {
+		return "", "", fmt.Errorf("error parsing github repository URL %q: %w", u, err)
+	}
+	path := strings.TrimPrefix(u.Path, "/")
+	path = strings.TrimSuffix(path, ".git")
+	parts := strings.Split(path, "/")
+	if len(parts) != 2 {
+		return "", "", fmt.Errorf("could not parse github repository URL %q", u)
```
Comment (hiddeco) [inline internal/gitprovider/github/github.go:157]:
```suggestion
		return "", "", fmt.Errorf("could not extract repository owner and name from URL %q", u)
```

maybe?

**Review by hiddeco — APPROVED** [review]
The suggestion is non-blocking.

**Comment by hiddeco** [comment]
Feels like this could potentially be easily backported to 0.7.1?

**Comment by krancour** [comment]
> Feels like this could potentially be easily backported to 0.7.1?

I wouldn't have done a v0.7.1 just for this, but since we're doing one for other reasons, I'm also inclined to include it.

File: internal/gitprovider/github/github.go (hunk 1 above)
Comment (krancour) [reply internal/gitprovider/github/github.go:157]: Yes. I like this wording better. Thanks!

**Pull request merged**